config/core/configmaps/autoscaler.yaml

```
@@ -172,3 +172,29 @@ data:
     # See the algorithm here: http://bit.ly/38XiCZ3.
     # TODO(vagababov): tune after actual benchmarking.
     activator-capacity: "100.0"
+
+    # initial-scale is the cluster-wide default value for the initial deployment
+    # replica size of a revision.
+    # This value cannot be set to 0 while allow-zero-initial-scale is false.
+    # If annotation "autoscaling.knative.dev/initialScale" is not set,
+    # or "autoscaling.knative.dev/initialScale" is set to 0 while
+    # allow-zero-initial-scale is false, the cluster wide initial-scale value will
+    # be used for the initial revision deployment size.
+    # If annotation "autoscaling.knative.dev/initialScale" is set to > 0,
+    # or 0 while allow-zero-initial-scale is true, the annotation value will be used
+    # for the initial revision deployment size.
+    initial-scale: "1"
+
+    # allow-zero-initial-scale is the cluster-wide flag to indicate whether
+    # initial-scale is allowed to be 0.
+    # If this is set to true, both the cluster-wide flag initial-scale and
+    # the annotation "autoscaling.knative.dev/initialScale" are allowed
+    # to be set to 0.
+    # If this is set to false, the cluster-wide initial-scale value is not
+    # allowed to be set to 0. If user attemps to deploy a new revision with
+    # "autoscaling.knative.dev/initialScale" annotation set to 0,
+    # the new revision will not be successfully created. If there's an existing
+    # revision with "autoscaling.knative.dev/initialScale" annotation
+    # set to 0, while allow-zero-initial-scale is changed from true to false,
+    # the default value 1 will be used.
```

I think this isn't quite true. If `allow-zero-initial-scale` is false and the annotation value is zero, [the cluster-wide default is used](https://github.com/knative/serving/blob/3a85ca83ba86c6f6d39b6b57f60fa4c7fbbc929d/pkg/reconciler/autoscaling/kpa/resources/decider.go#L87), which may be >1 if it was changed. (I'm not really sure how big an issue any of this really is since initial scale only applies once per revision anyway so "changing" initial scale for an existing revision only means something in some pretty small edge cases, but 🤷).

Also, stating both that true allows and false disallows seems redundant and this is already quite a lot of text. Maybe we could lose the "If this is set to true" paragraph and tighten the wording a bit, something like this (unless there's a nuance I missed)?

```suggestion
    # allow-zero-initial-scale controls whether either the cluster-wide initial-scale flag, 
    # or the "autoscaling.knative.dev/initialScale" annotation, can be set to 0.
    # If this value is changed from true to false, existing "autoscaling.knative.dev/initialScale" 
    # annotations with value 0 will be ignored.
```

---

Right, technically the cluster value will be used in this case of the cluster value changing from true to false, but since the initial scale only applies for new revisions. Yes sounds good, the `true` case is pretty self-explanatory so we can lose it.